cypress/support/cypress-commands/assertion.js

```diff
@@ -18,120 +18,72 @@
-        for (const [type, list] of Object.entries(exceptionMethods)) {
-          // Looking for the method and params in each list, if matched returning that exception method.
-          methodInExceptionList = list.find((obj) => {
-            if (obj.method && obj.method.toLowerCase() === method.toLowerCase()) {
-              if (obj.hasOwnProperty(CONSTANTS.PARAM)) {
-                return _.isEqual(obj.param, param);
-              } else {
-                return true;
-              }
-            }
-            return false;
-          });
+  (errorResponse, expectedContent, apiObject, isExceptionMethod) => {
+    expectedContent = isExceptionMethod
+      ? fetchErrorValidationObjectForExceptionMethod()
+      : expectedContent;
 
-          // If method is prsent in the list, exiting the loop.
-          if (methodInExceptionList) {
-            exceptionType = type;
-            break;
-          }
-        }
-
-        if (methodInExceptionList) {
-          switch (exceptionType) {
-            case CONSTANTS.NOT_AVAILABLE_METHODS:
-              errorType = CONSTANTS.NOT_AVAILABLE;
-              break;
-            case CONSTANTS.NOT_PERMITTED_METHODS:
-              errorType = CONSTANTS.NOT_PERMITTED;
-              break;
-            default:
-              errorType = CONSTANTS.NOT_SUPPORTED;
-          }
-        }
-
-        // Extracting the error content validation object from the environment variable.
-        if (
-          UTILS.getEnvVariable(CONSTANTS.ERROR_CONTENT_VALIDATIONJSON, false) &&
-          UTILS.getEnvVariable(CONSTANTS.ERROR_CONTENT_VALIDATIONJSON)[errorType]
-        ) {
-          const contentObject = UTILS.getEnvVariable(CONSTANTS.ERROR_CONTENT_VALIDATIONJSON)[
-            errorType
-          ];
-          resolve(contentObject);
-        } else {
-          reject(`Unable to find Error content validation object for ${errorType}`);
-        }
-      });
-    };
-
-    if (expectedContent === CONSTANTS.EXCEPTION_ERROR_OBJECT) {
-      cy.wrap(fetchErrorValidationObjectForExceptionmethod(method, param)).then((errorObject) => {
-        validateErrorObjects(errorObject);
-      });
+    //  If error object having type as "schemaOnly", skipping the content vallidation
+    if (expectedContent.type === CONSTANTS.SCHEMA_ONLY) {
+      return;
+    } // Doing custom validation when error object having type as "custom"
+    else if (expectedContent.type === CONSTANTS.CUSTOM) {
+      const additionalParams = { apiObject: apiObject, expectedContent: expectedContent };
+      cy.customValidation(expectedContent, additionalParams);
     } else {
       validateErrorObjects(expectedContent);
     }
 
     // Function to do content validation for the error objects.
-    function validateErrorObjects(errorSchemaObject) {
-      console.log(`Debug : errorSchemaObject ` + JSON.stringify(errorSchemaObject));
+    function validateErrorObjects(errorContentObject) {
```

Comment:
Below we just changed the variable name and 111 - 126 lines are removed because its not required since we passing the response in params